src/background/hypothesis-chrome-extension.js

```diff
@@ -219,11 +219,15 @@ export default function HypothesisChromeExtension({
     state.clearTab(tabId);
   }
 
-  // installs or uninstalls the sidebar from a tab when the H
-  // state for a tab changes
+  /**
+   * installs or uninstalls the sidebar from a tab when the H
+   * state for a tab changes
+   *
+   * @param {chrome.tabs.Tab} tab
+   */
   function updateTabDocument(tab) {
     // If the tab has not yet finished loading then just quietly return.
-    if (!state.getState(tab.id).ready) {
+    if (tab?.id === undefined || !state.getState(tab.id).ready) {
```

Comment:
When does this situation happen where `tab` is undefined? Is the calling code doing something wrong?

Reply:
When you query for a tab that doesn't exist:

```
chrome.tabs.get(1, a => console.log(a))
undefined
Unchecked runtime.lastError: No tab with id: 1.
```

The callback get executed and an error is raised.

Reply:
This is related to #320